pkg/cmd/grafana-server/commands/cli.go

```diff
@@ -44,22 +44,27 @@ func (e exitWithCode) Error() string {
 }
 
 func RunServer(opt ServerOptions) int {
+	serverFs := flag.NewFlagSet("server", flag.ContinueOnError)
 	var (
-		configFile = flag.String("config", "", "path to config file")
-		homePath   = flag.String("homepath", "", "path to grafana install/home path, defaults to working directory")
-		pidFile    = flag.String("pidfile", "", "path to pid file")
-		packaging  = flag.String("packaging", "unknown", "describes the way Grafana was installed")
-
-		v           = flag.Bool("v", false, "prints current version and exits")
-		vv          = flag.Bool("vv", false, "prints current version, all dependencies and exits")
-		profile     = flag.Bool("profile", false, "Turn on pprof profiling")
-		profileAddr = flag.String("profile-addr", "localhost", "Define custom address for profiling")
-		profilePort = flag.Uint64("profile-port", 6060, "Define custom port for profiling")
-		tracing     = flag.Bool("tracing", false, "Turn on tracing")
-		tracingFile = flag.String("tracing-file", "trace.out", "Define tracing output file")
+		configFile = serverFs.String("config", "", "path to config file")
+		homePath   = serverFs.String("homepath", "", "path to grafana install/home path, defaults to working directory")
+		pidFile    = serverFs.String("pidfile", "", "path to pid file")
+		packaging  = serverFs.String("packaging", "unknown", "describes the way Grafana was installed")
+
+		v           = serverFs.Bool("v", false, "prints current version and exits")
+		vv          = serverFs.Bool("vv", false, "prints current version, all dependencies and exits")
+		profile     = serverFs.Bool("profile", false, "Turn on pprof profiling")
+		profileAddr = serverFs.String("profile-addr", "localhost", "Define custom address for profiling")
+		profilePort = serverFs.Uint64("profile-port", 6060, "Define custom port for profiling")
+		tracing     = serverFs.Bool("tracing", false, "Turn on tracing")
+		tracingFile = serverFs.String("tracing-file", "trace.out", "Define tracing output file")
 	)
-
-	flag.Parse()
+	switch os.Args[1] {
+	case "server":
```

Comment:
here we actually don't need it if we don't want any sub-command in the future, just leave it to make the flag set usage more standard. I don't have strong opinion towards it, totally fine with a simple serverFs.Parse(os.Args[1:])

Reply:
Skip it for now. Taking height for something potentially happening in the future feels bad.